slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -185,6 +186,11 @@ std::string SLAMServiceImpl::PaintMap() {
         submap_poses =
             map_builder.map_builder_->pose_graph()->GetAllSubmapPoses();
     }
+    cartographer::transform::Rigid3d global_pose;
+    {
+        std::lock_guard<std::mutex> lk(viam_response_mutex);
+        global_pose = latest_global_pose;
+    }
```

Comment:
[q] Can we move this into the `if (marker_flag)` statement below, such that we grab the `latest_global_pose` only when we want to add the pose marker?

Reply:
sure, only thought is moving it out when we figure out how to have maps protected by this mutex as well

Reply:
ah, I see now why you have it here. I think that's fine then.